tests/profile.py

```diff
@@ -0,0 +1,584 @@
+    # ------------------------------------------------
+    job_parser = subparsers.add_parser(
+        "job",
+        parents=[common_parser],
+        description=("Profile the performance of your file system by processing files from the provided "
+                     "Conductor Job ID (jid)"),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+
+    job_parser.add_argument(
+        "jid",
+        help="The jid (job id) for a job whose files to target",
+    )
+
+    job_parser.set_defaults(func=run_job_profiler)
+
+    # ------------------------------------------------
+    # Directory parser
+    # ------------------------------------------------
+    dir_parser = subparsers.add_parser(
+        "dirs",
+        parents=[common_parser],
+        description=("Profile the performance of your file system by processing files found in the"
+                     "the given directories"),
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter
+    )
+
+    dir_parser.add_argument(
+        "dirs",
+        nargs="+",
+        help="One or more directories to read files from (recursively)",
+    )
+
+    dir_parser.set_defaults(func=run_dir_profiler)
+
+    return parser.parse_args()
+
+
+def cast_to_bool(string):
+    '''
+    Ensure that the argument provided is either "True" or "False (or "true" or
+    "false") and convert that argument to an actual bool value (True or False).
+    '''
+    string_lower = string.lower()
+    if string_lower == "true":
+        return True
+    elif string_lower == "false":
+        return False
+    raise argparse.ArgumentTypeError('Argument must be True or False')
+
+
+def run_job_profiler(args):
+
+    # ------------------------------------------------
+    # Fetch Job
+    # ------------------------------------------------
+    logger.info("Fetching Job %s", args.jid)
+    client = api_client.ApiClient()
+    r_body, _ = client.make_request('api/v1/jobs',
+                                    params="filter=jid_eq_%s" % args.jid,
+                                    use_api_key=True
+                                    )
+
+    jobs = json.loads(r_body).get("data")
+    if not jobs:
+        raise Exception("No Job %s found" % args.jid)
+    assert len(jobs) < 2, "Found more than 1 Job with jid %s" % args.jid
+    job = jobs[0]
+    upload_id = job["upload"]
+
+    # ------------------------------------------------
+    # Fetch Job's Upload
+    # ------------------------------------------------
+    logger.info("Fetching Upload %s", upload_id)
+    r_body, _ = client.make_request('api/v1/uploads/%s' % upload_id,
+                                    use_api_key=True
+                                    )
+    upload = json.loads(r_body).get("data")
+    filepaths = upload["upload_files"].keys()
+
+    profile(
+        filepaths,
+        stat=args.stat,
+        read=args.read,
+        hash_md5=args.md5,
+        hash_xx=args.xxhash if XXHASH else None,
```

Comment:
How does `argparse` behave if accessing a property for an argument that doesn't exist?

Since the `--xxhash` flag isn't enabled if `XXHASH` if `False`, I'm wondering if an exception would be raised if this code was run on a systems that's missing the `xxhash` module.

Reply:
If you expand that statement to use multiple lines, you can more easily see why it's not problematic, e.g,
```
if XXHASH:
     hash_xx=args.xxhash
else:
    hash_xx=None
```
In other words, the `arg.xxhash` value will not be accessed if xxhash is not installed.